### PR TITLE
Add check on tiered storage configuration and avoid NPE

### PR DESCRIPTION
**Pull request by apc999**
Fixes https://github.com/Alluxio/alluxio/issues/9020 and https://github.com/Alluxio/alluxio/issues/8997

When users specify a unknown storage alias (e.g., "ssd" rather than "SSD"), they will see the following error message on NPE:
```
2019-05-13 14:29:07,034 ERROR ProcessUtils - Uncaught exception while stopping Alluxio worker, simply exiting.
java.lang.NullPointerException
        at alluxio.worker.block.DefaultBlockWorker.stop(DefaultBlockWorker.java:262)
        at alluxio.Registry.stop(Registry.java:149)
        at alluxio.worker.AlluxioWorkerProcess.stopWorkers(AlluxioWorkerProcess.java:287)
        at alluxio.worker.AlluxioWorkerProcess.stop(AlluxioWorkerProcess.java:274)
        at alluxio.ProcessUtils.run(ProcessUtils.java:41)
        at alluxio.worker.AlluxioWorker.main(AlluxioWorker.java:69)
```

This PR makes alluxio service fail explicitly and early with the following error message:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at alluxio.proxy.AlluxioProxy.main(AlluxioProxy.java:44)
Caused by: java.lang.IllegalStateException: Alias "ssd" on tier 0 on worker (configured by alluxio.worker.tieredstore.level0.alias) is not found in global tiered storage setting: MEM,SSD,HDD
	at com.google.common.base.Preconditions.checkState(Preconditions.java:862)
	at alluxio.conf.InstancedConfiguration.checkTieredStorage(InstancedConfiguration.java:544)
	at alluxio.conf.InstancedConfiguration.validate(InstancedConfiguration.java:363)
	at alluxio.util.ConfigurationUtils.reloadProperties(ConfigurationUtils.java:443)
	at alluxio.conf.ServerConfiguration.reset(ServerConfiguration.java:68)
	at alluxio.conf.ServerConfiguration.<clinit>(ServerConfiguration.java:60)
	... 1 more
```

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 1]
```diff
@@ -520,6 +521,33 @@ private void checkTieredLocality() {
...
   }
 
+  /**
+   * Checks that tiered storage configuration on worker is consistent with the global configuration.
+   *
+   * @throws IllegalStateException if invalid tiered storage configuration is encountered
+   */
+  private void checkTieredStorage() {
+    int globalTiers = getInt(PropertyKey.MASTER_TIERED_STORE_GLOBAL_LEVELS);
+    Set<String> globalTierAliasSet = Sets.newHashSet();
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java]: This is deprecated in favor of `new HashSet<>()`

**Comment by apc999** [comment]
@calvinjia  PTAL

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**